addons/event/static/src/js/icon_selection_widget.js

```diff
@@ -0,0 +1,27 @@
+odoo.define('event.icon_selection_widget', function (require) {
+"use strict";
+
+var core = require('web.core');
+var registry = require('web.field_registry');
+var AbstractField = require('web.AbstractField');
+var QWeb = core.qweb;
+
+var IconSelectionWidget = AbstractField.extend({
```

Comment:
SelectionAsIcon ? SelectionIcon ?

Or simply IconSelection (like we already have LabelSelection).